### ISSUE TITLE
ionice not available on Solaris

### ISSUE THREAD

**Issue by krzaczek** [gitlabhq/gitlab_git#57]
Hi,,

This line breaks gitlab_git on solaris. ionice is not available on Solaris and should not be used the same way it's not used on "darwin" kernels.

https://github.com/gitlabhq/gitlab_git/blob/8192622ac919fb0c58eb4bc652d8785252a596f2/lib/gitlab_git/repository.rb#L932

```sh
>> Gem::Platform.local
=> #<Gem::Platform:0x30a78c @cpu="x86_64", @os="solaris2.11", @version=nil>
>> Gem::Platform.local.os
=> "solaris2.11"
```

```sh
[root@gitlab /home/git]# uname -a
SunOS gitlab 5.11 joyent_20131009T194502Z i86pc i386 i86pc Solaris
```

**Comment by randx** [comment]
@krzaczek can you propose a fix?

**Comment by krzaczek** [comment]
@randx  I'm not a rubby guy so I won't send You a PR ... but it looks like You need to modify line https://github.com/gitlabhq/gitlab_git/blob/8192622ac919fb0c58eb4bc652d8785252a596f2/lib/gitlab_git/repository.rb#L931 and exclude solaris the same way You exclude darwin kernels

**Comment by randx** [comment]
@maxlazio can you look into it?

**Comment by maxlazio** [comment]
Yes, MR is here: https://gitlab.com/gitlab-org/gitlab_git/merge_requests/25

**Comment by DouweM** [comment]
MR has been merged in 4ce7663c00670a7c3f7e5e6ce8d1f7541f51881b.
